frontend/src/pages/draft/index.tsx

```diff
@@ -134,14 +134,21 @@ export default function Draft({ editorAPI }: { editorAPI: EditorAPI }) {
 		updateSavedItems(newSaved);
 	}
 
+	// Update the cursor position
 	async function handleSelectionChanged() {
-		const docText = await getDocContext(true);
-		updateDocContext(docText);
-		const { charsToCursor, docLength } = await getCursorPosInfo();
+		// Get the document context (before cursor, selected text, after cursor)
+		const docInfo = await getDocContext();
+		updateDocContext(docInfo);
+
+		// Update the cursor position
+		const charsToCursor = docInfo.beforeCursor.length;
+    const docLength = docInfo.beforeCursor.length + docInfo.selectedText.length + docInfo.afterCursor.length;
```

Comment:
weird indentation?

Perhaps we actually want `updateCursorAtEnd((docInfo.selectedText + docInfo.afterCursor).trim().length > 0)`?